core/app/services/spree/cart/set_quantity.rb

```diff
@@ -0,0 +1,19 @@
+module Spree
+  module Cart
+    class SetQuantity
+      prepend Spree::ServiceModule::Base
+
+      def call(order:, line_item:, quantity: nil)
```

Comment:
@WitekL this should be wrapped in DB transaction

Reply:
@damianlegawiec Now I've changed code so that this service uses `failure` from the `ServiceModule`. Do we still need DB transaction?

Reply:
@WitekL still this needs to be wrapped in DB transaction for successful responses. If we update a line item and recalculate fails it will lead to corrupted order data (invalid totals, wrong promotions etc).